src/lib/Util/PermissionUtilInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Util;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+interface PermissionUtilInterface
```

Comment:
Utility class with an interface is bad by design. This is clearly more than a utility class thus it can be renamed to e.g. `PermissionChecker` class. Also I'd think about extracting some parts to other classes i.e. part related to user groups.